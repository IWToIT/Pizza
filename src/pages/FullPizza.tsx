import React from 'react';
import axios from 'axios';
import { useParams, useNavigate } from 'react-router-dom';

const FullPizza = () => {
  const [pizza, setPizza] = React.useState<{
    imageUrl: string;
    title: string;
    price: string;
  }>();
  
  const { id } = useParams();
  const navigate = useNavigate();

  React.useEffect(() => {
    async function fetchPizza() {
      try {
        const { data } = await axios.get('https://6474f3f57de100807b1bfd08.mockapi.io/pizzas/' + id);
        setPizza(data);
      } catch (error) {
        alert('Ошибка при получении пиццы!');
        navigate('/');
      }
    }

    fetchPizza();
  }, []);

  if (!pizza) {
    return 'Загрузка...';
  }

  return (
    <div className="container">
      <img src={pizza.imageUrl} alt={pizza.title} />
      <h2>{pizza.title}</h2>
      <h4>{pizza.price} ₽</h4>
    </div>
  );
};

export default FullPizza;
