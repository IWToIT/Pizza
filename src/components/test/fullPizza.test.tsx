import React from "react";
import axios from "axios";
import { render } from "./test-utils";
import { screen, waitFor } from "@testing-library/react";
import FullPizza from "../../pages/FullPizza";

test('отображается ли пицца при успешном получении данных', async () => {
  // Mock axios.get to return a successful response with pizza data
  jest.spyOn(axios, 'get').mockResolvedValue({ data: { imageUrl: 'testImageUrl', title: 'testTitle', price: 'testPrice' } });
  const { getByAltText, getByText } = render(<FullPizza />);

  await waitFor(() => expect(axios.get).toHaveBeenCalledTimes(1));

  expect(screen.getByTestId('container-element')).toBeInTheDocument();
  expect(screen.getByTestId('container-element')).toMatchSnapshot();
  expect(screen.getByAltText('testTitle')).toBeInTheDocument();
  expect(screen.getByText('testTitle')).toBeInTheDocument();
  expect(screen.getByText('testPrice ₽')).toBeInTheDocument();
});