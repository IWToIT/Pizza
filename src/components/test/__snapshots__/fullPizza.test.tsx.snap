// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображается ли пицца при успешном получении данных 1`] = `
<div
  class="container"
  data-testid="container-element"
>
  <img
    alt="testTitle"
    src="testImageUrl"
  />
  <h2>
    testTitle
  </h2>
  <h4>
    testPrice
     ₽
  </h4>
</div>
`;
